sdks/go/pkg/beam/x/hooks/perf/perf.go

```diff
@@ -0,0 +1,105 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package perf
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"runtime/pprof"
+
+	"github.com/apache/beam/sdks/go/pkg/beam/core/runtime"
+	"github.com/apache/beam/sdks/go/pkg/beam/core/runtime/harness"
+	fnpb "github.com/apache/beam/sdks/go/pkg/beam/model/fnexecution_v1"
+)
+
+var (
+	sessionCaptureHookKey = harness.HookKey("session_capture")
+	traceCaptureHookKey   = harness.HookKey("trace_capture")
+)
+
+// TraceCaptureHook is used by the harness to have the runner
+// persist a trace record with the supplied name and comment.
+// The type of trace can be determined by the prfix of the string
+// cpu: A profile compatible with traces produced by runtime/pprof
```

Comment:
<!--new_thread; commit:d3b9b8992429b98ae5429208961299a90dcdec50; resolved:0-->
To better call out this enumeration in the html go doc, consider adding a blank line and a bullet point for each of the trace and cpu lines, which will cause it to be in a preformatted box. As it's currently written, it'll be a single ongoing paragraph.

eg. 
// * cpu: A profile compatible with traces produced by runtime/pprof 
// * trace: A trace compatible with traces produced by runtime/trace

This should make it clearer in both rendered and text forms.

(The * doesn't matter. Any symbol can be used, as long as the intended block has a common prefix the renderer will do the right thing. See https://golang.org/pkg/go/doc/#ToHTML)

Reply:
Done.